pkg/kvstore/consul.go

```diff
@@ -601,13 +651,17 @@ func (c *consulClient) CreateOnly(ctx context.Context, key string, value []byte,
 }
 
 // createIfExists creates a key with the value only if key condKey exists
-func (c *consulClient) createIfExists(ctx context.Context, condKey, key string, value []byte, lease bool) error {
+func (c *consulClient) createIfExists(ctx context.Context, condKey, key string, value []byte, lease bool) (err error) {
 	// Consul does not support transactions which would allow to check for
 	// the presence of a conditional key if the key is not the key being
 	// manipulated
 	//
 	// Lock the conditional key to serialize all CreateIfExists() calls
 
+	defer func() {
+		Trace("CreateIfExists", err, logrus.Fields{fieldKey: key, fieldValue: string(value), fieldCondition: condKey, fieldAttachLease: lease})
+	}()
+
 	l, err := LockPath(ctx, condKey)
```

Comment:
You are converting other `:=` to `=` but not the ones in this function

Reply:
oops. The trace should have been added to the public function just below this. I've switched it out. This function no longer needs the named err return variable.